tree-sitter/src/TreeSitter/Unmarshal/Examples.hs

```diff
@@ -0,0 +1,62 @@
+{-# LANGUAGE DeriveAnyClass, DeriveGeneric, DuplicateRecordFields, TypeOperators #-}
+module TreeSitter.Unmarshal.Examples () where
+
+import qualified Data.Text as Text
+import GHC.Generics ((:+:), Generic)
+import TreeSitter.Unmarshal
+
+-- | An example of a sum-of-products datatype.
+data Expr a
+  = IfExpr (If a)
+  | BlockExpr (Block a)
+  | LitExpr (Lit a)
+  | BinExpr (Bin a)
+  deriving (Generic, Unmarshal)
+
+-- | Product with multiple fields.
+data If a = If { ann :: a, condition :: Expr a, consequence :: Expr a, alternative :: Maybe (Expr a) }
```

Comment:
These datatypes are intended to be representative, but not exhaustive. We employ sums, products, leaves, and anonymous leaves; as well as `Maybe`, `[]`, and `:+:`; but not `NonEmpty`.